File: day08/main.scala
object Day08 {
  def main(args: Array[String]): Unit = {
    val lines = scala.io.Source.fromFile("input").mkString.trim
    // val lines = "0222112222120000"
    val size = 25*6
    val split = lines.grouped(size).toArray

    val answer1 = split.map(s => (s.count(_ == '0'), s.count(_ == '1') * s.count(_ == '2') ))
      .minBy(_._1)

    println(answer1._2)

    val answer2 = 0.until(size).map(i => split.find(l => l(i) != '2').get(i))
    println(answer2.mkString.replace("0", " ").replace("1", "#").grouped(25).mkString("\n"))
  }
}

File: day08/main.rs
fn main() {
    let input = std::fs::read_to_string("input")
        .expect("Unable to read file");

    let size = 25*6;
    let chars: Vec<_> = input.trim().chars().collect();

    let min0 = chars.chunks(size)
        .min_by_key(|&w| w.iter().filter(|&s| *s == '0').count())
        .unwrap();

    let answer1 = min0.iter().filter(|&s| *s == '1').count()
        * min0.iter().filter(|&s| *s == '2').count();

    let decoded: Vec<_> = (0..size)
        .map(|x| chars.chunks(size).find(|&s| s[x] != '2').unwrap()[x])
        .map(|x| if x == '1' { "#" } else { " " })
        .collect();

    let answer2 = decoded.chunks(25)
        .map(|x| x.join(""))
        .collect::<Vec<String>>()
        .join("\n");

    println!("{}", answer1);
    println!("{}", answer2);
}
